wordpress-mobile/WordPress-iOS
# Use PostRepository.search in PostSearchService

There are a few issues in the "sync posts" function in `PostServices`, which I have described in https://github.com/wordpress-mobile/WordPress-iOS/pull/21814. This PR replaces the "sync posts" function call in the searching post feature with the new search function in `PostRepository`.

## Test Instructions

Verify posts search works as expected on both Posts List and Pages List, including filtering by tag and auto loading when scrolling to the bottom of the search result list.

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
The "Test Instructions" section.

3. What automated tests I added (or what prevented me from doing so)
None.

PR submission checklist:


UI Changes testing checklist: N/A
